optuna/integration/mlflow.py

```diff
@@ -85,73 +157,42 @@ def objective(trial):
             set, key value pairs in study.user_attrs will supersede existing tags.
     """
 
-    def __init__(
-        self,
-        tracking_uri: Optional[str] = None,
-        metric_name: str = "value",
-        nest_trials: bool = False,
-        tag_study_user_attrs: bool = False,
-    ) -> None:
-
-        _imports.check()
-
-        self._tracking_uri = tracking_uri
-        self._metric_name = metric_name
-        self._nest_trials = nest_trials
-        self._tag_study_user_attrs = tag_study_user_attrs
-
     def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -> None:
 
-        # This sets the tracking_uri for MLflow.
-        if self._tracking_uri is not None:
-            mlflow.set_tracking_uri(self._tracking_uri)
-
-        # This sets the experiment of MLflow.
-        mlflow.set_experiment(study.study_name)
+        self.set_experiment(study)
 
         with mlflow.start_run(run_name=str(trial.number), nested=self._nest_trials):
 
             # This sets the metric for MLflow.
-            trial_value = trial.value if trial.value is not None else float("nan")
-            mlflow.log_metric(self._metric_name, trial_value)
+            if trial.value:
```

Comment:
I feel this breaks a (minor) backward compatibility. Is it necessary this `if` statement? The tests seem fine without this line.

Reply:
MLFlow does not like None as input for log_metric, if I remember right. I write a test for that (and if it is not necessary, remove the line)

Reply:
I just checked, and the objective function actually should always return a float, hence we should be fine without this line. (Or actually a sequence of floats -> make it compatible with both.

Reply:
I updated the code of `log_metric` such that it can handle `float`, `Sequence[float]` and `None`